api/pkg/s3/listbuckets.go

```diff
@@ -58,10 +58,12 @@ func parseListBuckets(list *s3.ListBucketsResponse) ListBucketsResponse {
 }
 
 func (s *APIService) ListBuckets(request *restful.Request, response *restful.Response) {
-	if !policy.Authorize(request, response, "bucket:list") {
+	log.Infof("Received request for all buckets")
+	err := signature.PayloadCheck(request, response)
```

Comment:
Trying to understand the changes. Payload check is for API authorization?
How the multi-cloud API signature validation works in multi-cloud s3?
Can this payload check be part of the signature validation?

Reply:
See [this issue](https://github.com/sodafoundation/multi-cloud/issues/920) please. Payload checking need to read all content.